### ISSUE TITLE
Returning address of variable copied into lambda produces spurious warning.

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#17748]
|  |  |
| --- | --- |
| Bugzilla Link | [17374](https://llvm.org/bz17374) |
| Version | 3.2 |
| OS | All |
| Reporter | LLVM Bugzilla Contributor |
| CC | @DougGregor |

## Extended Description 
Given the following sample code:

int main()
{
    int i;
    auto f = [=]()mutable->int*
    {
            return &i;
    };

    return 0;
}

The compiler reports "address of stack memory associated with local variable 'i' returned" for the 'i' in the lambda. That warning would be legitimate for a plain function but not for a lambda.

Note that this bug appears similar to bug #&#8203;16548 but it is not the same. Here, the variable is captured by value rather than by reference.

**Comment by wheatman** [comment]
This was fixed between 3.4 and 3.5
https://godbolt.org/z/dvjTEWhjd

code
```
//compile with -std=c++11 -Wreturn-stack-address
int main()
{
    int i = 0;
    auto f = [=]()mutable->int*
    {
            return &i;
    };

    return 0;
}
```

old warning
```
<source>:7:21: warning: address of stack memory associated with local variable 'i' returned [-Wreturn-stack-address]
            return &i;
                    ^
1 warning generated.
Compiler returned: 0
```
